streaming/base/download.py

```diff
@@ -94,6 +94,32 @@ def download_from_sftp(remote: str, local: str) -> None:
     os.rename(local_tmp, local)
 
 
+def download_from_gcs(remote: str, local: str) -> None:
+    """Download a file from remote GCS to local.
+
+    Args:
+        remote (str): Remote path (S3).
+        local (str): Local path (local filesystem).
+    """
+    import boto3
+    from botocore.exceptions import ClientError
+
+    obj = urllib.parse.urlparse(remote)
+    if obj.scheme != 'gs':
+        raise ValueError(f'Expected obj.scheme to be "gs", got {obj.scheme} for remote={remote}')
+
+    gcs_client = boto3.client('s3',
```

Comment:
lol the client is instantiated as an s3 client?

Reply:
Yes, that's how boto3 supports it. Important part is `endpoint_url='https://storage.googleapis.com'`